DEV-INTRO.md

```diff
@@ -1,59 +1,166 @@
+
+The compiled code is written to `/usr/local/lib/R/site-library` which is mapped to `volumes/lib` on your machine. This same folder is mounted in many other containers, allowing you to share the same PEcAn modules in all containers. Now if you change a module, and compile all other containers will see and use this new version of your module.
 
+To compile the PEcAn code you can use the make command in either the rstudio container, or in the executor container. The script [`compile.sh`](sripts/compile.sh) will run make inside the executor container.
+
+### Workflow Submission
+
+You can submit your workflow either in the executor container or in rstudio container. For example to run the `docker.sipnet.xml` workflow located in the tests folder you can use: 
+
+```
+docker-compose exec executor bash
+# inside the container
+cd /pecan/tests
+R CMD ../web/workflow.R docker.sipnet.xml
 ```
-./scripts/build.sh -install # installs all R packages
-./scripts/build.sh -h       # list other options
 
-```
\ No newline at end of file
+A better way of doing this is developed as part of GSOC, in which case you can leverage of the restful interface defined, or using the new R PEcAn API package.
+
+# Directory Structure
+
+Following are the main folders inside the pecan repository. 
+
+### base (R packages)
+
+These are the core packages of PEcAn. Most other packages will depend on the packages in this folder.
+
+### models (R packages)
+
+Each subfolder contains the required pieces to run the model in PEcAn
+
+### modules (R packages)
+
+Contains packages that either do analysis, or download and convert different data products.
+
+### web (PHP + javascript)
+
+The Pecan web application
+
+### shiny (R + shiny)
+
+Each subfolder is its own shiny application.
+
+### book_source (RMarkdown)
+
+The PEcAn documentation that is compiled and uploaded to the PEcAn webpage.
+
+### docker
+
+Some of the docker build files. The Dockerfiles for each model are placed in the models folder.
+
+### scripts
+
+Small scripts that are used as part of the development and installation of PEcAn.
+
+# Advanced Development Options
+
+## docker-compose
+
+To make it easier to start the containers and not having to remember to use `docker-compose -f docker-compose.yml -f docker-compose.dev.yml` when you want to use the docker-compose comannd, you can rename `docker-compose.dev.yml` to `docker-compose.override.yml`. The docker-compose command will automatically use the `docker-compose.yml`, `docker-compose.override.yml` and the `.env` files to start the right containers with the correct parameters.
```

Comment:
I think it will be safer to put the renaming again to the beginning and get rid of all these `-f` flags

I think it will take someone longer to have issues because they forgot they had extra configurations in the override, I mean that feels like a bit more advanced level of developing than just running the stack